### PR TITLE
move to python pod serving up ignition instead of S3

### PR DESCRIPTION
**Pull request by joelddiaz**
Rather than set up an S3 bucket to provide the ignition config to the worker nodes, set up a Pod on the hypershift host to serve up ignition.

Remove the S3 bucket creation/deletion code as it is no longer needed.

### CODE REVIEW

File: pkg/aws/install.go [hunk 1]
```diff
@@ -537,6 +553,162 @@ func InstallCluster(name, releaseImage, dhParamsFile string, waitForReady bool)
...
+	}
+
+	route := &routev1.Route{}
+	route.APIVersion = "v1"
+	route.Name = "ignition-provider"
+	route.Spec = routev1.RouteSpec{
+		Port: &routev1.RoutePort{
+			TargetPort: intstr.IntOrString{
+				Type:   intstr.String,
+				StrVal: "https",
```
Comment (csrwng) [inline pkg/aws/install.go]: s/https/http/

File: pkg/aws/install.go [hunk 2]
```diff
@@ -537,6 +553,162 @@ func InstallCluster(name, releaseImage, dhParamsFile string, waitForReady bool)
...
+	}
+
+	service := &corev1.Service{}
+	service.APIVersion = "v1"
+	service.Name = "ignition-provider"
+	service.Spec = corev1.ServiceSpec{
+		Type: corev1.ServiceTypeClusterIP,
+		Ports: []corev1.ServicePort{
+			{
+				Name:     "https",
```
Comment (csrwng) [inline pkg/aws/install.go]: s/https/http/

File: pkg/aws/install.go [hunk 3]
```diff
@@ -537,6 +553,162 @@ func InstallCluster(name, releaseImage, dhParamsFile string, waitForReady bool)
...
+				Protocol: corev1.ProtocolTCP,
+				Port:     443,
```
Comment (csrwng) [inline pkg/aws/install.go]: I would leave this as 8000 or 80... given that the target is not TLS, just plain-text http

File: pkg/aws/install.go [hunk 4]
```diff
@@ -537,6 +553,162 @@ func InstallCluster(name, releaseImage, dhParamsFile string, waitForReady bool)
...
+	route.APIVersion = "v1"
+	route.Name = "ignition-provider"
+	route.Spec = routev1.RouteSpec{
+		Port: &routev1.RoutePort{
+			TargetPort: intstr.IntOrString{
+				Type:   intstr.String,
+				StrVal: "https",
+			},
+		},
+		TLS: &routev1.TLSConfig{
```
Comment (csrwng) [inline pkg/aws/install.go]: I would remove the TLS bit unless you plan to include the CA in the ignition file. This should be a plain-text http for now. Later on we'll need to think harder about security.

**Review by csrwng — COMMENTED** [review]
@joelddiaz this is awesome. Some comments about the service/route.
For the future we need to think how to secure this payload given that it can be used to add arbitrary workers to the cluster.
Also (for a follow-up) we may want to stop generating manifests in code and follow the pattern in hypershift-toolkit where we have templated manifests as assets and just render those. It will make things easier to maintain going forward.

File: pkg/aws/install.go (hunk 1 above)
Comment (joelddiaz) [reply pkg/aws/install.go]: i'm just going to remove the ports as it will just use everything (all one ports) defined in the Service

**Comment by joelddiaz** [comment]
@csrwng i moved to yaml files that we can more easily update, and just sore them with the go-bindata tool. Also fixed up the https/tls bits that you suggested. PTAL, and i'll squash

**Comment by csrwng** [comment]
lgtm
ship it!

**Comment by joelddiaz** [comment]
sqashed, so merging

**Pull request merged**